NASA-PDS/registry-api-service
# JSON PDS4 Label

See https://github.com/NASA-PDS/pds-api/issues/101

**Build Instructions**
- Build `pds-api-javalib` from "swaggerhub-pds_federated_api-pds-api-101" branch.
- Build and install `pds-api-javalib` locally (`mvn install`. You may need to remove "maven-gpg-plugin" from pom.xml if the build fails.)
- Build `registry-api-service`

**Create test data**
- Use the latest Harvest and Registry Manager builds.
- Delete old and create new registry.
- Harvest and load some data into Elasticsearch.
- Latest Harvest version creates `ops:Label_File_Info/ops:json_blob` field (aka JSON BLOB) by default. This field contains PDS4 label in JSON format (compressed and Base64 encoded).
- Make sure that documents in Elasticsearch have JSON BLOB:
  `http://localhost:9200/registry/_search?q=*&_source=ops:Label_File_Info/ops:json_blob`

**Test API Server**
- Start API server (e.g., `java -jar registry-api-service-0.4.0-SNAPSHOT.jar`)
- Go to Swagger UI (`localhost:8080`)
- Select the endpoint you want to test (bundle, collection, product)
- Select the `application/pds4+json` response content type from the dropdown
- Click "Try it out" button
- You should see response similar to this:

```
{
  "summary": {
    "start": 0,
    "limit": 100
  },
  "data": [
    {
      "id": "urn:nasa:pds:kaguya_grs_spectra::1.1",
      "meta": {
        "node_name": "PDS_ENG",
        "label_file": {
          "file_name": "bundle_kaguya_derived.xml",
          "file_ref": "/C:/ws3/Geo/lunar/urn-nasa-pds-kaguya_grs_spectra/bundle_kaguya_derived.xml",
          "creation_date": "2020-11-06T21:09:09Z",
          "file_size": "8556",
          "md5_checksum": "eabfe35431bcfbfc0faa2ced241398dc"
        }
      },
      "pds4": {
        "Product_Bundle": {
          "Identification_Area": {
            "product_class": "Product_Bundle",
            "Modification_History": {
              "Modification_Detail": {
                "modification_date": "2019-09-30",
                "description": "Peer reviewed version of Kaguya Gamma-Ray Spectrometer Corrected Spectra PDS4",
                "version_id": 1
              }
            },
...
```
